In [3]:
#사용되는 라이브러리
#scipy : t검정에 사용

from scipy import stats
import numpy as np

In [12]:
#학생 키에 대한리스트 생성하기
height_list = np.array([169, 167, 175, 166, 162, 180, 172
                       , 162, 173, 162, 181, 175, 181, 181
                       , 162, 165, 172, 176, 167, 165])

### t-검정

In [13]:
tTestResult = stats.ttest_1samp(height_list,  170)

In [14]:
tTestResult
#statistic : 검정 통계량
#pvalue(p-value) = 유의수준
#170값 : 귀무가설에서 채택한 값
#      "집단의 키 평균은 170이다"

#대립가설 : 집단의 키 평균은 170이 아니야
#t- 검정의 목적 : 대립가설이 참인지 확인하는 것
#확인은 => p-value<0.05채택(글로벌스탠다드)
           p-value>=0.05 기각
#채택과 기각은 대립가설을 기준으로 함

Ttest_1sampResult(statistic=0.42270173526659166, pvalue=0.6772582597305596)

In [15]:
print("t검정통계량 = %.3f, p-value = %.3f" %(tTestResult))

t검정통계량 = 0.423, p-value = 0.677


해석 1
statistic = 검정통계량 = 정확도 <br>
pvalue(p-value) = 유의수준 = 오차

정확도 + 오차 =1

해석2

해당집단이 키 170이라는 귀무가설에 대해<br>
검정 통계량 42.3% , p 값이 0.677%으로<br>
유의수준이 표준편차±2.5를 상회하는 수치임으로<br>
해당 가설의 기무가설은 기각됨을 알 수 있다<br>
따라서 해당 집단의 키 평균은 170이 아님을 알 수 있다


---

## 두집단

In [ ]:
import numpy as np
from scipy import stats

In [26]:
group1Heights = np.array([162, 168, 169, 165, 166, 168
                          , 162, 172, 157, 173, 158, 169
                          , 164, 170, 163, 175, 177, 162
                         , 175, 177])
group2Heights = np.array([180, 181, 163, 164, 174, 169
                           , 164, 172, 162, 171, 180, 168
                           , 164, 169, 178, 177,167, 179
                           , 172, 169])

#두집단간 평균비교
tTestResult = stats.ttest_ind(group1Heights, group2Heights)
tTestResult

Ttest_indResult(statistic=-1.8253194633881713, pvalue=0.07582039848129221)

In [35]:
beforeWeights =np.array([80, 82, 76, 82, 73, 77
                       , 78, 61, 81, 80, 70, 60, 83
                       , 89, 84, 85, 81, 67, 60])

#20개의 랜덤값 생성하기
temp_list = []
for i in beforeWeights :
    #랜덤한 가중치부여
    temp_list.append( i * np.random.normal(0.99, 0.02) )

afterWeights = np.array(temp_list)

tTestResult = stats.ttest_rel(beforeWeights, afterWeights)
tTestResult

Ttest_relResult(statistic=2.009126013353649, pvalue=0.05976259236283378)